runtime/onert/api/include/nnfw.h

```diff
@@ -257,6 +257,37 @@ NNFW_STATUS nnfw_prepare(nnfw_session *session);
  */
 NNFW_STATUS nnfw_run(nnfw_session *session);
 
+/**
+ * @brief     Run inference asynchronously
+ *
+ * <p>This function must be called after model is loaded by {@link nnfw_load_model_from_file},
+ * session is prepared for inference by {@link nnfw_prepare}, set input and output buffers
+ * by {@link nnfw_set_input} and {@link nnfw_set_output}.</p>
+ *
+ * <p>This function returns immediately after starting a thread to run the inference.
+ * To get the result of it {@nnfw_run_wait} must be called</p>
+ *
+ * @param[in] session The session to run inference
+ * @return    @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_run_async(nnfw_session *session);
+
+/**
+ * @brief     Wait for asynchronous run to finish
+ *
+ * <p>This function must be called after calling {@link nnfw_run_asnyc}, and can be called only once
+ * for a {@link nnfw_run_async} call.
+ *
+ * <p>When this function returns, it means that this session has finished
+ * the asynchronous run. Then the user can safely use the output data.</p>
+ *
+ * <p>This function return after the asynchronous inference is finished.</p>
+ *
+ * @param[in] session The session to run inference
+ * @return    @c NNFW_STATUS_NO_ERROR if successful
+ */
+NNFW_STATUS nnfw_run_await(nnfw_session *session);
```

Comment:
shout the name be `nnfw_await`? `nnfw_run_await` has two verbs in name.

Reply:
Hmm... that sounds better I guess